feat: add tolgee translator article + documentation (#421)
tolgee/documentation: blog/2023-06-06-tolgee-translator.md

tags: [translation, chatgpt, in-context, sdk, i18n]
---

A question frequently asked is whether AI will replace human translators. AI has made its way into various aspects of our lives, including localization, where it demonstrates exceptional capabilities. However, there are two significant challenges hindering the complete adoption of AI for software translation: reliability and context. Tolgee translator, an experimental tool, aims to tackle these challenges head-on.

![Tolgee translator](/img/blog/tolgee-translator/post-ai.png)

<!-- truncate -->

## The concept

Upon trying our machine translation tools, one quickly realizes that they struggle with short, simple labels. Surprisingly, they perform better when handling longer texts. Why is that? Context.

Without context, how can one accurately translate a word like "Save"? It could refer to saving a form or saving money. Without contextual information, the translation service can only guess. Moreover, while we may understand that the translation pertains to software, a general translation service lacks this awareness. Hence, context is key.

> For a long time, we pondered the idea that if we could provide AI with the same context as a human translator, it could potentially deliver comparable translation quality. However, no translation services existed that could effectively incorporate context.

Reviewer: ```suggestion
For a long time, we considered that providing AI with the same context as a human translator could deliver comparable translation quality. However, no translation services existed that could effectively incorporate context.
```

## So, what's new in Tolgee?

To enhance our tool, the latest version of Tolgee SDK collects data from your application. Rest assured, this data does not include any personal or sensitive information. We simply send a list of keys present on the page. Using this data, Tolgee analyzes the order of the keys to establish statistical relationships between them.

When utilizing the Tolgee translator, we identify the most relevant keys and present them to ChatGPT, along with Translation Memory data (similar to how a human translator would work).

## Results

Based on our observations in the Czech language, the context has proven to be quite helpful. It assists in determining word meanings and plays a crucial role in consistent terminology usage. For example, the translation of the term "namespace" into Czech can be awkward, so we decided to stick with the English term. However, traditional translation services would always attempt to translate it. ChatGPT, on the other hand, understands our approach when provided with an example, ensuring consistency without relying solely on a glossary.

### What about reliability?

> Our primary intention behind releasing the Tolgee translator is to provide assistance to human translators. Therefore, the tool is designed to support them in the translation process, requiring their review and input. However, this setup allows us to gradually empower AI and potentially reduce the need for extensive human interaction in the future.

Reviewer: Would be great to add some CTA at the end of every article, to boost an engagement, or offer help to create a long-term relationship.

Here is an example from some of those:
> 💡 Do you have any suggestions on how to make the localization process with Tolgee better? [Let us know](mailto:info@tolgee.io) or create a pull request on [GitHub](https://github.com/tolgee/tolgee-platform).

In case of this article, you can e.g. invite the readers to join our Slack community (link: https://tolg.ee/slack). If they have any question regarding the AI translator, or a specific question related to their project, they can discuss it there :)
